lib/media/stream.js

```diff
@@ -274,9 +274,13 @@ shaka.media.Stream.prototype.resync = function() {
  * Resync the stream.
  *
  * @param {boolean} forceClear
+ * @param {number=} opt_clearBufferOffset if |forceClear| and |opt_clearBufferOffset|
+ *     are truthy, clear the stream buffer from the offset (in front of video
+ *     currentTime) to the end of the stream.
+ *
  * @private
  */
-shaka.media.Stream.prototype.resync_ = function(forceClear) {
+shaka.media.Stream.prototype.resync_ = function(forceClear, opt_clearBufferOffset) {
```

Comment:
Please assert somewhere that |opt_clearBufferOffset|, if set, is non-negative as a sanity check.